app/jinja_filters.py

```diff
@@ -188,10 +188,21 @@ def setAttributes(dictionary, attributes):
 
 @blueprint.app_template_filter()
 def should_wrap_with_fieldset(question):
-    answers = question["answers"]
+    # Logic for when to wrap with a fieldset comes from
+    # https://ons-design-system.netlify.app/patterns/question/
+    if question["type"] == "DateRange":
+        return False
 
-    if len(answers) > 1 and not any(
-        answer["type"] in {"Date", "MonthYearDate", "Duration"} for answer in answers
+    answers = question["answers"]
+    if (
+        question["type"] == "MutuallyExclusive"
+        or len(answers) > 1
+        or (
+            len(answers) == 1
+            and answers[0]["type"]
+            in ["Radio", "Date", "MonthYearDate", "Duration", "Address", "Relationship"]
```

Comment:
This covers the scenario where there is only one answer and no answer label defined in the schema. In that scenario the question text should be used for the legend. 

There are two distinct types of answers that this covers:
- options answers - radio and relationships
- multi field answers - dates, durations and address

DateRange is handled separately as it would otherwise fall into the `len(answers) > 1` logic.